### DISCUSSION TITLE
v1.21.0 - LocalAI functions!

### DISCUSSION THREAD

**Discussion by mudler** [mudler/LocalAI#760] [Announcements]
### Bug fixes :bug:
* fix: make first stream message to send empty content by @mudler in https://github.com/go-skynet/LocalAI/pull/751
### Exciting New Features 🎉
* feat: [whisper] Partial support for verbose_json format in transcribe endpoint by @ldotlopez in https://github.com/go-skynet/LocalAI/pull/721
* feat: LocalAI functions by @mudler in https://github.com/go-skynet/LocalAI/pull/726
* feat: `gRPC`-based backends by @mudler in https://github.com/go-skynet/LocalAI/pull/743
* feat: falcon support (7b and 40b) with `ggllm.cpp` by @mudler in https://github.com/go-skynet/LocalAI/pull/743
### Other Changes
* :arrow_up: Update nomic-ai/gpt4all by @ci-robbot in https://github.com/go-skynet/LocalAI/pull/704
* fix(deps): bump go-llama.cpp by @mudler in https://github.com/go-skynet/LocalAI/pull/719
* Create FUNDING.yml by @mudler in https://github.com/go-skynet/LocalAI/pull/725
* fix(deps): update module github.com/sashabaranov/go-openai to v1.13.0 by @renovate in https://github.com/go-skynet/LocalAI/pull/667
* fix(deps): update module github.com/otiai10/openaigo to v1.5.2 by @renovate in https://github.com/go-skynet/LocalAI/pull/731
* fix(deps): update github.com/tmc/langchaingo digest to a875e6b by @renovate in https://github.com/go-skynet/LocalAI/pull/700
* fix(deps): update github.com/go-skynet/go-llama.cpp digest to ffa57fb by @renovate in https://github.com/go-skynet/LocalAI/pull/707
* fix(deps): update github.com/nomic-ai/gpt4all/gpt4all-bindings/golang digest to d611d10 by @renovate in https://github.com/go-skynet/LocalAI/pull/699
* Update README with sponsors section by @mudler in https://github.com/go-skynet/LocalAI/pull/732
* fix(deps): update github.com/go-skynet/go-ggml-transformers.cpp digest to 3fec197 by @renovate in https://github.com/go-skynet/LocalAI/pull/706
* :arrow_up: Update nomic-ai/gpt4all by @ci-robbot in https://github.com/go-skynet/LocalAI/pull/705
* fix(deps): update module google.golang.org/protobuf to v1.31.0 by @renovate in https://github.com/go-skynet/LocalAI/pull/746
* fix(deps): update module github.com/sashabaranov/go-openai to v1.14.0 by @renovate in https://github.com/go-skynet/LocalAI/pull/739
* fix(deps): update module gopkg.in/yaml.v2 to v3 by @renovate in https://github.com/go-skynet/LocalAI/pull/299
* fix(deps): update github.com/go-skynet/go-ggml-transformers.cpp digest to ffb09d7 by @renovate in https://github.com/go-skynet/LocalAI/pull/744
* :arrow_up: Update donomii/go-rwkv.cpp by @ci-robbot in https://github.com/go-skynet/LocalAI/pull/750
* fix(deps): update github.com/mudler/go-ggllm.cpp digest to 862477d by @renovate in https://github.com/go-skynet/LocalAI/pull/745
* fix(deps): update github.com/tmc/langchaingo digest to dcf7ecd by @renovate in https://github.com/go-skynet/LocalAI/pull/736
* fix(deps): update github.com/nomic-ai/gpt4all/gpt4all-bindings/golang digest to cfd70b6 by @renovate in https://github.com/go-skynet/LocalAI/pull/734
* fix(deps): update github.com/go-skynet/go-llama.cpp digest to 6c97625 by @renovate in https://github.com/go-skynet/LocalAI/pull/733
* :arrow_up: Update nomic-ai/gpt4all by @ci-robbot in https://github.com/go-skynet/LocalAI/pull/735
* :arrow_up: Update go-skynet/go-ggml-transformers.cpp by @ci-robbot in https://github.com/go-skynet/LocalAI/pull/711
* fix(deps): update github.com/go-skynet/go-bert.cpp digest to 6abe312 by @renovate in https://github.com/go-skynet/LocalAI/pull/756
* fix(deps): update module github.com/gofiber/fiber/v2 to v2.48.0 by @renovate in https://github.com/go-skynet/LocalAI/pull/757

## New Contributors
* @ldotlopez made their first contribution in https://github.com/go-skynet/LocalAI/pull/721

**Full Changelog**: https://github.com/go-skynet/LocalAI/compare/v1.20.1...v1.21.0

<hr /><em>This discussion was created from the release <a href='https://github.com/go-skynet/LocalAI/releases/tag/v1.21.0'>v1.21.0 - LocalAI functions!</a>.</em>

**Comment by tmc** [comment]
Great release! Can you point me to how to run the falcon 7b model via grpc?

**Reply by rioncarter** [reply]
Were you able to find a way to run Falcon? My attempts all seem to end with:

> "message": "rpc error: code = Unknown desc = unimplemented"

**Reply by djmaze** [reply]
Same here.
